app/main/autoUpdate.ts

```diff
@@ -0,0 +1,143 @@
+import axios, { AxiosRequestConfig } from 'axios';
+import { ipcMain, BrowserWindow } from 'electron';
+import { autoUpdater, UpdateInfo } from 'electron-updater';
+import log from 'electron-log';
+import { createHash, createVerify, Verify } from 'crypto';
+import fs from 'fs';
+
+import {
+    updateAvailable,
+    updateDownloaded,
+    updateVerified,
+    updateError,
+} from '~/constants/ipcRendererCommands.json';
+import {
+    triggerAppUpdate,
+    quitAndInstallUpdate,
+} from '~/constants/ipcCommands.json';
+import { publicKeyUrl } from '~/constants/verification.json';
+
+import { version } from '../package.json';
+import { build } from '../../package.json';
+
+const updateServer = 'https://update.electronjs.org';
+const updateFeed = `${updateServer}/${build.publish.owner}/${build.publish.repo}/${process.platform}-${process.arch}/${version}/`;
+autoUpdater.setFeedURL({
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    ...(build.publish as any),
+    url: updateFeed,
+});
+
+log.transports.file.level = 'info';
+
+autoUpdater.logger = log;
+autoUpdater.autoDownload = false;
+autoUpdater.autoInstallOnAppQuit = false;
+
+interface RealUpdateInfo extends UpdateInfo {
+    downloadedFile: string;
+}
+
+async function getRemoteContent(url: string, binary = false) {
+    const options: AxiosRequestConfig = {};
+
+    if (binary) {
+        options.responseType = 'arraybuffer';
+    }
+
+    const { data } = await axios.get(url, options);
+    return data;
+}
+
+function getSha256Sum(path: string): Promise<string> {
+    const sum = createHash('sha256');
+    const stream = fs.createReadStream(path);
+
+    stream.on('data', (data) => sum.update(data));
+
+    return new Promise((resolve, reject) => {
+        stream.on('end', () => resolve(sum.digest('hex')));
+        stream.on('error', reject);
+    });
+}
+
+function getVerifier(path: string): Promise<Verify> {
+    const verifier = createVerify('RSA-SHA256');
+    const stream = fs.createReadStream(path);
+
+    stream.on('data', (data) => verifier.update(data));
+
+    return new Promise((resolve, reject) => {
+        stream.on('end', () => resolve(verifier));
+        stream.on('error', reject);
+    });
+}
+
+const releasesFeed = `https://github.com/${build.publish.owner}/${build.publish.repo}/releases/download`;
+
+function getVerificationFunctions(
+    { downloadedFile: filePath, releaseName, path: fileName }: RealUpdateInfo // UpdateInfo interface doesn't seem to be aligned with actual content.
+) {
+    const releaseFileUrl = `${releasesFeed}/${releaseName}/${fileName}`;
+
+    return {
+        async verifyChecksum() {
+            const remoteHash = (
+                await getRemoteContent(`${releaseFileUrl}.hash`)
```

Comment:
I think we should use a file format extension like `.sha256` or `.sha256sum` to indicate what type of hash is contained in the file.